src/niweb/apps/noclook/helpers.py

```diff
@@ -123,23 +139,33 @@ def form_update_node(user, handle_id, form, property_keys=None):
         for field in form.base_fields.keys():
             if field not in meta_fields:
                 property_keys.append(field)
-    for key in property_keys:
-        if form.cleaned_data.get(key, None) or form.cleaned_data.get(key, None) == 0:
-            pre_value = node.data.get(key, '')
-            if pre_value != form.cleaned_data[key]:
-                node.data[key] = form.cleaned_data[key]
-                # Handle dates
-                if hasattr(form.cleaned_data[key], 'isoformat'):
-                    node.data[key] = form.cleaned_data[key].isoformat()
 
-                if key == 'name':
-                    nh.node_name = form.cleaned_data[key]
-                activitylog.update_node_property(user, nh, key, pre_value, form.cleaned_data[key])
-        elif form.cleaned_data.get(key, None) == '' and key in node.data.keys():
-            if key != 'name':  # Never delete name
-                pre_value = node.data.get(key, '')
-                del node.data[key]
-                activitylog.update_node_property(user, nh, key, pre_value, form.cleaned_data[key])
+    # Validate prerequisites to avoid runtime errors
+    if not (hasattr(form, 'cleaned_data') and form.cleaned_data and hasattr(node, 'data') and isinstance(node.data, dict)):
+        raise ValueError("Invalid form or node data structure")
```

Comment:
Hmm would it not be better to do one for each and tell what is actually wrong...?